includes/abstracts/calendar.php

```diff
@@ -570,11 +570,17 @@ public function set_time_format( $format = '' ) {
 	 */
 	public function set_datetime_separator( $separator = '' ) {
 
+		$separator_spacing = get_post_meta( $this->id, '_calendar_datetime_separator_spacing', true );
 		if ( empty( $separator ) ) {
 			$separator = get_post_meta( $this->id, '_calendar_datetime_separator', true );
 		}
 
-		$this->datetime_separator = esc_attr( $separator );
+		if ( empty( $separator_spacing ) ) {
+			$separator = '&nbsp;' . trim( $separator ) . '&nbsp;';
+		} else {
+			$separator = str_replace( ' ', '&nbsp;', $separator );
+		}
+		$this->datetime_separator = wp_kses_post( $separator );
```

Comment:
Something I just thought of: is this function preparing this data to be saved in the DB or to be output on the screen? If it's for output, shouldn't we be late escaping instead?